components/automate-cli/pkg/verifyserver/services/systemuserservice/systemuserservice.go

```diff
@@ -0,0 +1,211 @@
+	}
+	su.Log.Debug("Group 'hab' found successfully")
+	return successResponse(constants.SYSTEM_GROUP_HAB_VALIDATION_SUCCESS_TITLE, constants.SYSTEM_GROUP_HAB_SUCCESS_MSG)
+}
+
+func (su *SystemUserService) ValidateHabUserAndGroupMapping() *models.Checks {
+	// Check if "hab" user is a member of "hab" group
+	userInGroup := su.checkUserInGroup(constants.USERNAME, constants.GROUPNAME)
+
+	if !userInGroup {
+		su.Log.Error("Validation failed: User 'hab' is not a member of group 'hab'")
+		return failureResponse(constants.SYSTEM_USERANDGROUP_MAPPING_FAILURE_TITLE, constants.SYSTEM_USERINGROUP_ERROR_MSG, constants.SYSTEM_USERANDGROUP_MAPPING_RESOLUTION_MSG)
+	}
+	su.Log.Debug("User 'hab' is a member of group 'hab'")
+
+	// Check if "hab" user's primary group is "hab" group
+	primaryGroupMatch := su.checkPrimaryGroupMatch(constants.USERNAME, constants.GROUPNAME)
+
+	if !primaryGroupMatch {
+		su.Log.Error("Validation failed: Primary group mapping for user 'hab' is not 'hab' group")
+		return failureResponse(constants.SYSTEM_USERANDGROUP_MAPPING_FAILURE_TITLE, constants.SYSTEM_PRIMARYGROUP_MATCH_ERROR_MSG, constants.SYSTEM_USERANDGROUP_MAPPING_RESOLUTION_MSG)
+	}
+
+	su.Log.Debug("Validation passed: Primary group mapping for user 'hab' is 'hab' group")
+	return successResponse(constants.SYSTEM_USERANDGROUP_MAPPING_SUCCESS_TITLE, constants.SYSTEM_USERANDGROUP_MAPPING_SUCCESS_MSG)
+}
+
+// Check if "hab" user exists
+func (su *SystemUserService) isHabUserPresent(username string) bool {
+	_, err := su.user.Lookup(username)
+	if err != nil {
+		su.Log.Error("Hab user not present", err)
+		return false
+	}
+	su.Log.Debug("Hab user exists")
+	return true
+}
+
+// Check if "hab" group exists
+func (su *SystemUserService) isHabGroupPresent(groupname string) bool {
+	_, err := su.user.LookupGroup(groupname)
+	if err != nil {
+		su.Log.Error("Hab group not present", err)
+		return false
+	}
+	su.Log.Debug("Hab group exists")
+	return true
+}
+
+// Delete a user
+func (su *SystemUserService) deleteUser(username string) error {
+	_, err := su.exec.Command(constants.USERDELCMD, []string{username})
+	return err
+}
+
+// Check if a user is a member of a group
+func (su *SystemUserService) checkUserInGroup(username, groupname string) bool {
+	output, err := su.exec.Command("id", []string{"-Gn", username})
+	if err != nil {
+		su.Log.Error("Command execution failed to check group membership of user", err)
+		return false
+	}
+
+	groups := strings.Split(strings.TrimSpace(string(output)), " ")
+	for _, group := range groups {
+		if group == groupname {
+			su.Log.Debug("User is a member of group")
+			return true
+		}
+	}
+	su.Log.Debug("User is not a member of group")
+	return false
+}
+
+// Check if a user's primary group matches a specified group
+func (su *SystemUserService) checkPrimaryGroupMatch(username, groupname string) bool {
```

Comment:
This function is just checking that the user's primary group is present in list of groups or not. This does not seem right.
We want to to ensure that the `hab` group is primary group for user `hab`

Reply:
Done